### PR TITLE
federate vrops and snmp metrics seperately

### PR DESCRIPTION
**Pull request by artherd42**

### CODE REVIEW

File: system/prometheus-infra/templates/_prometheus-infra.yaml.tpl [hunk 1]
```diff
@@ -1,3 +1,116 @@
...
+    - source_labels: [__name__, ifIndex, server_id]
+      regex: '^snmp_[a-z0-9]*_if.+;(.+);(.+)'
+      replacement: '$1@$2'
+      target_label: uniqueident
+      action: replace
+    - source_labels: [__name__, instance]
+      regex: '^probe_success;(.+)'
+      replacement: '$1'
+      target_label: target
+      action: replace
```
Comment (richardtief) [inline system/prometheus-infra/templates/_prometheus-infra.yaml.tpl:38]: These relabelings are specific to `snmp` and `cloud prober` metrics and can be removed for this job.

File: system/prometheus-infra/templates/_prometheus-infra.yaml.tpl [hunk 2]
```diff
@@ -1,3 +1,116 @@
...
```
Comment (richardtief) [inline system/prometheus-infra/templates/_prometheus-infra.yaml.tpl:100]: Here also. `cloud prober` metric relabelings can be removed.

**Review by richardtief — CHANGES_REQUESTED** [review]
Looks good. Only one small thing.

File: system/prometheus-infra/templates/_prometheus-infra.yaml.tpl [hunk 3]
```diff
@@ -1,3 +1,116 @@
...
+    - regex: "prometheus_replica|kubernetes_namespace|kubernetes_name|namespace|pod|pod_template_hash|instance"
+      action: labeldrop
```
Comment (viennaa) [inline system/prometheus-infra/templates/_prometheus-infra.yaml.tpl:40]:
```suggestion
```

Adding the suggestion to Richards comment

File: system/prometheus-infra/templates/_prometheus-infra.yaml.tpl [hunk 4]
```diff
@@ -1,3 +1,116 @@
...
```
Comment (viennaa) [inline system/prometheus-infra/templates/_prometheus-infra.yaml.tpl:102]:
```suggestion
```

**Review by viennaa — CHANGES_REQUESTED** [review]
Adding removal suggestions to @richardtief comments.

Furthermore, this block can be removed from the overall `prometheus-infra-collector` job

https://github.com/sapcc/helm-charts/blob/master/system/prometheus-infra/templates/_prometheus-infra.yaml.tpl#L93-L97

```
    - source_labels: [__name__, ifIndex, server_id]
      regex: '^snmp_[a-z0-9]*_if.+;(.+);(.+)'
      replacement: '$1@$2'
      target_label: uniqueident
      action: replace
```

**Review by viennaa — APPROVED** [review]

**Review by richardtief — APPROVED** [review]
Thanks for taking over.

**Pull request merged**